iroha/src/asset.rs

```diff
@@ -6,99 +6,33 @@ use parity_scale_codec::{Decode, Encode};
 pub struct Asset {
     /// identifier of asset, formatted as asset_name#domain_id
     pub id: Id,
-    pub amount: u128,
+    pub quantity: u128,
 }
 
 impl Asset {
     pub fn new(id: Id) -> Self {
-        Asset { id, amount: 0 }
+        Asset { id, quantity: 0 }
     }
 
-    pub fn with_amount(mut self, amount: u128) -> Self {
-        self.amount = amount;
+    pub fn with_quantity(mut self, quantity: u128) -> Self {
+        self.quantity = quantity;
         self
     }
 }
 
 pub mod isi {
     use super::*;
-    use crate::isi::Contract;
-    use iroha_derive::{IntoContract, Io};
-    use parity_scale_codec::{Decode, Encode};
+    use std::ops::Add;
 
     /// The purpose of add asset quantity command is to increase the quantity of an asset on account of
     /// transaction creator. Use case scenario is to increase the number of a mutable asset in the
     /// system, which can act as a claim on a commodity (e.g. money, gold, etc.).
-    #[derive(Clone, Debug, PartialEq, Io, IntoContract, Encode, Decode)]
-    pub struct AddAssetQuantity {
-        pub asset_id: Id,
-        pub account_id: Id,
-        pub amount: u128,
-    }
-
-    impl Instruction for AddAssetQuantity {
-        fn execute(&self, world_state_view: &mut WorldStateView) -> Result<(), String> {
-            let assets = &mut world_state_view
-                .account(&self.account_id)
-                .ok_or("Account not found.")?
-                .assets;
-            if let Some(asset) = assets.get_mut(&self.asset_id) {
-                asset.amount += self.amount;
-            } else {
-                assets.insert(self.asset_id.clone(), Asset::new(self.asset_id.clone()));
-                assets
-                    .get_mut(&self.asset_id)
-                    .ok_or("Failed to get asset.")?
-                    .amount += self.amount;
-            }
-            Ok(())
-        }
-    }
+    impl Add<u128> for Asset {
```

Comment:
Maybe use type alias for `u182` for readability? E.g. `Add<Quantity> for Asset`

Reply:
Also should decide how to name it - somewhere it's amount - somewhere quantity

Reply:
https://soramitsu.atlassian.net/browse/RH2-88